### PR TITLE
Refactor Dockerfile

### PR DESCRIPTION
**Pull request by lfdebrux**
### What problem does this pull request solve?

This PR makes a few tweaks to the Dockerfile to make it easier to use with Docker Compose and the update_app_versions script in forms-deploy.

The changes here should be similar to those in alphagov/forms-admin#905.

### Things to consider when reviewing

- Ensure that you consider the wider context.
- Does it work when run on your machine?
- Is it clear what the code is doing?
- Do the commit messages explain why the changes were made?
- Are there all the unit tests needed?
- Has all relevant documentation been updated?

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -342,6 +344,7 @@ GEM
     zeitwerk (2.6.12)
 
 PLATFORMS
+  aarch64-linux-musl
```
Comment (DavidBiddle) [inline Gemfile.lock:347]: This line / file isn't changed in the other Dockerfile refactor PRs - should it be?

Comment (lfdebrux) [reply Gemfile.lock:347]: You got me :D I've been finding that whenever I run `docker compose build` this line gets added to my `Gemfile.lock` file, so I thought I would save me having to stash that every time by sneaking it in 😅 

In the other repos the platform `aarch64-linux` has been present for 2 years now (see forms-runner for example [[1](https://github.com/alphagov/forms-runner/blame/c2744035f052c866ee0ab313688d8a4db0bc3eaf/Gemfile.lock#L466)]. I'm not sure why this repo got missed.

Or, now that I look at it, why for forms-api it's `aarch64-linux-musl` and not just `aarch64-linux` 🤔

**Review by DavidBiddle — APPROVED** [review]
Code looks good and builds for me locally

**Comment by DavidBiddle** [comment]
Rebased to fix the merge conflict, but will now need someone else to approve please 🙏

**Review by thomasiles — APPROVED** [review]

**Pull request merged**